doc/release/compatibility.rst

```diff
@@ -0,0 +1,150 @@
+The same compatibility is maintained between :term:`release series` as well.
+
+An attempt to use a new feature results in one of the options:
+
+*   The attempt is successful.
+
+*   There is a meaningful error about the old data layout until the database schema upgrade.
+    It does not lead to a service outage or data corruption.
+    An instance can upgrade the data layout using the :ref:`box.schema.upgrade() <admin-upgrades>` call
+    to enable all new release features (when all instances of the replicaset are run on the same Tarantool version).
+
+..  _cg_binary_protocol:
+
+Binary protocol
+---------------
+
+The binary protocol evolves without breaking compatibility with old clients.
+All binary protocol requests operational in an older release should work on a newer one.
+They do not change the meaning.
+Responses have the same format, but mappings may contain fields not present in the older release.
+
+``net.box`` client of the older release is able to work
+with the newer one, except the features introduced in the newer release.
+``net.box`` client of the newer release is fully operational with the server
+running under the older one, except the features not implemented in the older release.
+
+..  _cg_replication_protocol:
+
+Replication protocol
+--------------------
+
+An instance run on a newer release may work as:
+
+*   upstream (master) of an instance with an older release
+
+*   downstream (replica) without a database schema upgrade.
+
+The database schema upgrade (``box.schema.upgrade()``) must be performed when all replicaset instances
+run on the same Tarantool version.
+The upgrade does not cause downtime if the application does not lean on internal schema representation.
+
+..  _cg_lua_code:
+
+Lua code
+--------
+
+If a code is run on an older release, it will operate with the same effect on a
+newer one. However, only meaningful code counts.
+If any code throws an error but starts doing something useful, the change is considered compatible.
+
+A room for new functionality is still here: adding new options, more
+fields to a returning table, and more returning values (multireturn).
+
+Adding a new built-in module or a new global value is considered as the compatible change.
+
+Adding a new field to an existing metatable is okay if it is not listed in the Lua 5.1 Reference Manual. Otherwise, it should be proven that it may not break any meaningful code.
+
+Examples of compatible changes:
+
+*   Add ``__pairs``, ``__ipairs`` to a metatable of a userdata.
+    It is not from Lua 5.1, and the userdata has no default behaviour for ``pairs()`` and `ipairs()` calls.
+
+*   Add ``__lt``, ``__le`` metamethod
+    (if the attempt to use ``<``, ``<=`` and so on leads to an error before the change).
+
+Examples of NOT compatible changes:
+
+*   Add ``__pairs``, ``__ipairs`` to a metatable of a table or cdata
+    (it already has its own behavior before the change).
```

Comment:
```suggestion
*   Add ``__pairs``, ``__ipairs`` to a metatable of a table or cdata
    (it already has a defined behavior before the change).
```
I think this might work